src/components/implementation/capmgr/naive/mem_mgr.c

```diff
@@ -65,3 +65,35 @@ memmgr_shared_page_map_cserialized(vaddr_t *pgaddr, int *unused, int idx)
 done:
 	return num_pages;
 }
+
+/* These functions return the address at which the gs register is set to */
+void *
+memmgr_tls_alloc(unsigned int dst_tid)
+{
+	/* Just return the vaddr range for this tid */
+	assert(dst_tid < MAX_NUM_THREADS);
```

Comment:
We should differentiate from `VALIDATING INPUT PARAMETERS` from the capmgr internal data consistencies.. So if `dst_tid` passed is invalid, we'd not want an assert here, instead return a failure..